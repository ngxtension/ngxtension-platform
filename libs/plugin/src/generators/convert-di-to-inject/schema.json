{
	"$schema": "http://json-schema.org/schema",
	"$id": "",
	"title": "",
	"type": "object",
	"properties": {
		"project": {
			"type": "string",
			"aliases": ["name", "projectName"],
			"description": "Project for which to convert from constructor DI to inject function."
		},
		"path": {
			"type": "string",
			"description": "Path to the component/directive/pipe you want to convert to use the inject function"
		},
		"includeReadonlyByDefault": {
			"type": "boolean",
			"aliases": ["includeReadonly", "includeReadonlyByDefault"],
			"description": "Whether to include readonly for injections by default. Default is false."
		},
		"useEsprivateFieldNotation": {
			"type": "boolean",
			"aliases": [
				"useESPrivateFieldNotation",
				"useEsprivateFieldNotation",
				"useEsPrivateFieldNotation"
			],
			"description": "Whether to replace TS private modifier with ES private field notation. Default is false."
		}
	}
}
