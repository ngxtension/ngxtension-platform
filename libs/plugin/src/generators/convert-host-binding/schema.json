{
	"$schema": "http://json-schema.org/schema",
	"$id": "ConvertHostBinding",
	"title": "",
	"type": "object",
	"properties": {
		"project": {
			"type": "string",
			"aliases": ["name", "projectName"],
			"description": "Project for which to convert from host binding to host property."
		},
		"path": {
			"type": "string",
			"description": "Path to the component/directive you want to convert to use the host property."
		}
	}
}
