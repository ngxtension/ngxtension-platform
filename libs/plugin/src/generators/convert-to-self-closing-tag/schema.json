{
	"$schema": "http://json-schema.org/schema",
	"$id": "",
	"title": "",
	"type": "object",
	"properties": {
		"project": {
			"type": "string",
			"aliases": ["name", "projectName"],
			"description": "Project for which to convert to self closing tags."
		},
		"path": {
			"type": "string",
			"description": "Path to the component/directive you want to convert to self closing tags"
		}
	}
}
