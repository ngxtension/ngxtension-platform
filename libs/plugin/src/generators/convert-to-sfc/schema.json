{
	"$schema": "http://json-schema.org/schema",
	"$id": "",
	"title": "",
	"type": "object",
	"properties": {
		"project": {
			"type": "string",
			"aliases": ["name", "projectName"],
			"description": "Project for which to convert to single file components."
		},
		"path": {
			"type": "string",
			"description": "Path to the component to convert to a single file component."
		},
		"maxInlineTemplateLines": {
			"type": "number",
			"description": "If the template has more lines than this, it will be skipped.",
			"default": 200
		},
		"maxInlineStyleLines": {
			"type": "number",
			"description": "If the styles have more lines than this, they will be skipped.",
			"default": 200
		},
		"moveStyles": {
			"type": "boolean",
			"description": "Move styles to the single file component.",
			"default": false
		}
	}
}
