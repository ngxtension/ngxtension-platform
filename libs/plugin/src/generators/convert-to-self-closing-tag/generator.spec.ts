import { Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';

import convertToSelfClosingTagGenerator from './generator';
import { ConvertToSelfClosingTagGeneratorSchema } from './schema';

const template = `
<div>Hello</div>
<app-my-cmp1>123</app-my-cmp1>
<app-my-cmp1>123</app-my-cmp1>
<app-my-cmp2 test="hello">123</app-my-cmp2>
<app-my-cmp3
  test="hello">
  123
</app-my-cmp3>
<app-my-cmp4
  test="hello"
>
  123
</app-my-cmp4>
<app-my-cmp5
  test="hello"
>
  123
</app-my-cmp5>
<app-my-cmp10 test="hello"
  [test]="hello"
  (test)="hello()"
>
</app-my-cmp10>

<app-my-cmp11
  test="hello"
  [test]="hello"
  (test)="hello()"
>
</app-my-cmp11>
<app-my-cmp12 test="hello"
>


</app-my-cmp12>
<input type="text" />

<app-my-cmp6 />
<app-my-cmp7 test="hello" />

<hello-world></hello-world>

<pagination count="1" [test]="hello" (test)="test"></pagination>

<pagination count="1" />

<hello-world></hello-world>

<hello-world12>
  <hello-world13>
    <hello-world14 count="1" [test]="hello" (test)="test" ></hello-world14>
      <hello-world15>
        <hello-world16  count="1" [test]="hello" (test)="test"  />
        <hello-world17  count="1" [test]="hello" (test)="test" ></hello-world17>
        <hello-world18
         count="1" [test]="hello"
          (test)="test"
          >

        </hello-world18>
      </hello-world15>
  </hello-world13>
</hello-world12>

<app-management
  *ngIf="
    categoryList &&
    ((test1 && test1.length > 0) ||
    (test && test.length > 0))
  "
  [test]="test > 2"
  [test]="test"
  (testEvent)="test.length > 0 ? test($event) : null"
  (testEvent2)="test1($event)"></app-management>
`;

const filesMap = {
	notComponent: `
import { Injectable } from '@angular/core';

@Injectable()
export class MyService {}
`,
	componentNoTemplate: `
import { Component } from '@angular/core';

@Component({})
export class MyCmp {}
`,

	componentWithTemplateUrl: `
import { Component, Input } from '@angular/core';

@Component({
  templateUrl: './my-file.html'
})
export class MyCmp {
}
`,

	componentWithInlineTemplate: `
import { Component, Input } from '@angular/core';

@Component({
  template: \`
    <router-outlet></router-outlet>
  \`
})
export class MyCmp {
}
`,
} as const;

describe('convertToSelfClosingTagGenerator', () => {
	let tree: Tree;
	const options: ConvertToSelfClosingTagGeneratorSchema = {
		path: 'libs/my-file.ts',
	};

	function setup(file: keyof typeof filesMap) {
		tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
		tree.write('package.json', `{"dependencies": {"@angular/core": "17.1.0"}}`);
		tree.write(`libs/my-file.ts`, filesMap[file]);

		if (file === 'componentWithTemplateUrl') {
			tree.write(`libs/my-file.html`, template);
			return () => {
				return [
					tree.read('libs/my-file.ts', 'utf8'),
					filesMap[file],
					tree.read('libs/my-file.html', 'utf8'),
					template,
				];
			};
		}

		return () => {
			return [tree.read('libs/my-file.ts', 'utf8'), filesMap[file]];
		};
	}

	it('should not do anything if not component/directive', async () => {
		const readContent = setup('notComponent');
		await convertToSelfClosingTagGenerator(tree, options);
		const [updated, original] = readContent();
		expect(updated).toEqual(original);
	});

	it('should not do anything if no template', async () => {
		const readContent = setup('componentNoTemplate');
		await convertToSelfClosingTagGenerator(tree, options);
		const [updated, original] = readContent();
		expect(updated).toEqual(original);
	});

	it('should convert properly for templateUrl', async () => {
		const readContent = setup('componentWithTemplateUrl');
		await convertToSelfClosingTagGenerator(tree, options);
		const [updated, , updatedHtml] = readContent();
		expect(updated).toMatchSnapshot();
		expect(updatedHtml).toMatchSnapshot();
	});

	it('should convert properly for inline template', async () => {
		const readContent = setup('componentWithInlineTemplate');
		await convertToSelfClosingTagGenerator(tree, options);
		const [updated, , updatedHtml] = readContent();
		expect(updated).toMatchSnapshot();
		expect(updatedHtml).toMatchSnapshot();
	});
});
