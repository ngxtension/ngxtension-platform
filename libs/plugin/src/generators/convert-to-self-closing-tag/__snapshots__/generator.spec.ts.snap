// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertToSelfClosingTagGenerator should convert properly for inline template 1`] = `
"
import { Component, Input } from '@angular/core';

@Component({
  template: \`
    <router-outlet  />
  \`
})
export class MyCmp {
}
"
`;

exports[`convertToSelfClosingTagGenerator should convert properly for inline template 2`] = `undefined`;

exports[`convertToSelfClosingTagGenerator should convert properly for templateUrl 1`] = `
"
import { Component, Input } from '@angular/core';

@Component({
  templateUrl: './my-file.html'
})
export class MyCmp {
}
"
`;

exports[`convertToSelfClosingTagGenerator should convert properly for templateUrl 2`] = `
"
<div>Hello</div>
<app-my-cmp1>123</app-my-cmp1>
<app-my-cmp1>123</app-my-cmp1>
<app-my-cmp2 test="hello">123</app-my-cmp2>
<app-my-cmp3
  test="hello">
  123
</app-my-cmp3>
<app-my-cmp4
  test="hello"
>
  123
</app-my-cmp4>
<app-my-cmp5
  test="hello"
>
  123
</app-my-cmp5>
<app-my-cmp10 test="hello"
  [test]="hello"
  (test)="hello()"
 />

<app-my-cmp11 test="hello"
  [test]="hello"
  (test)="hello()"
 />
<app-my-cmp12 test="hello"
 />
<input type="text" />

<app-my-cmp6 />
<app-my-cmp7 test="hello" />

<hello-world  />

<pagination count="1" [test]="hello" (test)="test"></pagination>

<pagination count="1" />

<hello-world  />

<hello-world12>
  <hello-world13>
    <hello-world14 count="1" [test]="hello" (test)="test"  />
      <hello-world15>
        <hello-world16  count="1" [test]="hello" (test)="test"  />
        <hello-world17 count="1" [test]="hello" (test)="test"  />
        <hello-world18 count="1" [test]="hello"
          (test)="test"
           />
      </hello-world15>
  </hello-world13>
</hello-world12>

<app-management *ngIf="
    categoryList &&
    ((test1 && test1.length > 0) ||
    (test && test.length > 0))
  "
  [test]="test > 2"
  [test]="test"
  (testEvent)="test.length > 0 ? test($event) : null"
  (testEvent2)="test1($event)" />
"
`;
